src/number.jl

```diff
@@ -14,8 +14,12 @@ scalartype(::Type{T}) where {T<:Number} = T
 #-----------------
 # note: required to make scale(NaN, 0) = 0
 @inline scale(x::Number, α::Number) = (iszero(α) ? zero(x) : x) * α
+@inline function scale(y::Number, x::Number, α::Number)
```

Comment:
I don't think this 3-arg `scale` method makes sense, so I would just take this as the definition for `scale!!`

Reply:
I agree, but it's a bit annoying because the natural definition for `promote_scale(y, x, a)` uses this. I can also turn it around, and make `scale!!(y::Number, x::Number, a::Number)` call `promote_scale(y, x, a)` and define the type promotion in that function. which might be better?

Reply:
Ok yes, I see. I guess having this 3-arg `scale` is not too bad after all. Then at least everything is nicely homogeneous in how it is implemented.